python/lsst/pipe/tasks/insertFakeSources.py

```diff
@@ -0,0 +1,597 @@
+            starImages.append((starIm.convertF(), xy))
+
+        return starImages
+
+    def cleanCat(self, fakeCat):
+        """Remove rows from the fakes catalog which have HLR = 0 for either the buldge or disk component
+
+        Parameters
+        ----------
+        fakeCat : `pandas.core.frame.DataFrame`
+
+        Returns
+        -------
+        fakeCat : `pandas.core.frame.DataFrame`
+        """
+
+        goodRows = ((fakeCat[self.config.bulgeHLR] != 0.0) & (fakeCat[self.config.diskHLR] != 0.0))
+
+        badRows = len(fakeCat) - len(goodRows)
+        self.log.info("Removing %d rows with HLR = 0 for either the bulge or disk" % badRows)
+
+        return fakeCat[goodRows]
+
+    def addFakeSources(self, calexp, fakeImages, sourceType):
+        """Add the fake sources to the given calexp
+
+        Parameters
+        ----------
+        fakeCat : `pandas.core.frame.DataFrame`
+        calexp : `lsst.afw.image.exposure.exposure.ExposureF`
+        fakeImages : `list`
+            A list of `lsst.afw.image.image.image.ImageF`
+
+        Returns
+        -------
+        calexp : `lsst.afw.image.exposure.exposure.ExposureF`
+
+        Notes
+        -----
+        Uses the x, y information in the ``fakeCat`` to position an image of the fake interpolated onto the
+        pixel grid of the image. Sets the ``FAKE`` mask plane for the pixels added with the fake source.
+        """
+
+        calexpBBox = calexp.getBBox()
+        calexpMI = calexp.maskedImage
+
+        for (fakeImage, xy) in fakeImages:
+            X0 = xy.getX() - fakeImage.getWidth()/2 + 0.5
+            Y0 = xy.getY() - fakeImage.getHeight()/2 + 0.5
+            self.log.debug("Adding fake source at %d, %d" % (xy.getX(), xy.getY()))
+            if sourceType == "galaxy":
+                interpFakeImage = afwMath.offsetImage(fakeImage, X0, Y0, "lanczos3")
+                interpFakeImBBox = interpFakeImage.getBBox()
+            else:
+                interpFakeImage = fakeImage
+                interpFakeImBBox = fakeImage.getBBox()
+
+            interpFakeImBBox.clip(calexpBBox)
+            calexpMIView = calexpMI.Factory(calexpMI, interpFakeImBBox)
+
+            if interpFakeImBBox.getArea() > 0:
+                clippedFakeImage = interpFakeImage.Factory(interpFakeImage, interpFakeImBBox)
+                clippedFakeImageMI = afwImage.MaskedImageF(clippedFakeImage)
+                clippedFakeImageMI.mask.set(self.bitmask)
+                calexpMIView += clippedFakeImageMI
+
+        return calexp
+
+    def processFakeCalexp(self, dataRef, calexp):
+        """Process the calexp now that fakes have been added.
+
+        Parameters
+        ----------
+        dataRef : `lsst.daf.persistence.butlerSubset.ButlerDataRef`
+        calexp : `lsst.afw.image.exposure.exposure.ExposureF`
+
+        Notes
+        -----
+        Overwrites the data productsproduced by processCcd except for the calexp. Background subtraction is
+        redone.
```

Comment:
I am concerned that you are overwriting all the data except the calexp, especially the background model, because that means if someone goes to use the calexp, then all the associated data products will not be what went into making that calexp or what was measured on that calexp (I.E. the background model wont be quite the same)